File: 백준/Bronze/1362. 펫/펫.py
import sys
input = sys.stdin.readline
cnt = 0
while True:
    state = False
    cnt += 1
    o,w = map(int,input().split())
    if o == 0 and w == 0: break
    while True:
        x,y = input().rstrip('\n').split()
        y = int(y)
        if x == '#': break
        elif x == 'E': w -= y
        else: w += y
        if w <= 0: state = True
    
    if w <= 0 or state: print("%d RIP"%cnt)
    elif o//2 < w < o*2: print("%d :-)"%cnt)
    else:print("%d :-("%cnt)

File: 백준/Bronze/1362. 펫/펫.java
import java.io.*;
import java.util.*;


public class Main {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        Long cnt = 0L;

        while (true) {
            Boolean state = false;
            cnt++;
            StringTokenizer st1 = new StringTokenizer(br.readLine());
            Integer o = Integer.parseInt(st1.nextToken());
            Integer w = Integer.parseInt(st1.nextToken());
            if (o == 0 && w == 0) break;
            while (true) {
                StringTokenizer st2 = new StringTokenizer(br.readLine());
                String x = st2.nextToken();
                if (x.equals("#")) break;
                else if (x.equals("E")) w -= Integer.parseInt(st2.nextToken());
                else w += Integer.parseInt(st2.nextToken());
                if (w <= 0) state = true;
            }

            if (w <= 0 || state) System.out.println(cnt + " "+ "RIP");
            else if (o/2 < w && w < o*2) System.out.println(cnt + " "+ ":-)");
            else System.out.println(cnt + " "+ ":-(");
        }
    }
}